src/plait/core.clj

```diff
@@ -25,15 +25,21 @@
                          [n (get new-bindings-m n b)]))
                 (remove (comp base-bindings-s first) new-binding-pairs)))))
 
-(def ^:dynamic *plait-bindings* [])
+(defn plait-impl [bindings x]
+  (if (and (seqable? x)
+           (= 'plait (first x)))
+    (let [child-bindings (rename-underscore-bindings (second x))
+          child-body (drop 2 x)
+          new-bindings (merge-bindings bindings child-bindings)]
+      (plait-impl new-bindings
+                  (concat ['let new-bindings]
+                          (map (partial plait-impl new-bindings) child-body))))
+    x))
```

Comment:
Hmm I think this breaks if you do this:

```
(plait [a 1
          b (inc a)]
  (do
    (plait [a 2]
      b)))
```

as it does not recur on the `do`?